### ISSUE TITLE
Freezing versions sometimes crashes, due to missing recipe repositories.

### ISSUE THREAD

**Issue by sykora** [raxod502/straight.el#348]
This might be a duplicate of an existing issue, but I have no idea how to attribute the errors I'm getting to a root cause.

If I load a version lockfile that contains a specified version for a recipe repository (e.g. `epkgs`), but none of the packages in my init file are actually installed from `epkgs` (e.g. `magit`, `multiple-cursors`), then `straight-freeze-versions` crashes, because there is no `epkgs` repository.

```
$ cd .../.emacs.d/straight/repos/epkgs/
$ git rev-parse HEAD

[Program not found]
```

Just the act of interactively calling `straight-use-package` subsequently, forces the clone of `epkgs`, following which I can freeze again successfully.

This seems to be traceable to the following in `straight-freeze-versions`:

```
           (unless (or (null local-repo)
                       (assoc local-repo versions-alist)
                       (straight--repository-is-available-p recipe))
             (straight-use-package (intern package)))))))
```
which apparently makes the assumption that if `local-repo` is in the `versions-alist`, then the recipe must have an available repository.

Is this a bug? It would seem that freezing versions should be idempotent. I've attached the init file I've used, just in case.

[init.txt](https://github.com/raxod502/straight.el/files/2815492/init.txt)

Thanks, and apologies if this is a spurious issue!

**Comment by raxod502** [comment]
Yes, you have identified two bugs in `straight.el`.

* When the `default-directory` for a command does not exist, `straight.el` erroneously reports `Program not found`, which is misleading. That is fixed in https://github.com/raxod502/straight.el/commit/7e1f7ad1b5367324bde2936a491ab91d3e3828e6. Now it will report `File error while setting current directory`.
* `straight.el` fails to clone recipe repositories while freezing versions, hence why later code that expected `epkgs` to be cloned was failing. This is fixed in https://github.com/raxod502/straight.el/commit/aab9d5e2cce618ce39f3d5dd918c713b9510c44f.

Now, you may have another complaint, namely that `straight.el` shouldn't have to clone `epkgs` in the first place. I agree, but fixing it is less straightforward—see #161.

**Comment by raxod502** [comment]
This thread is being closed automatically by [Tidier](https://github.com/raxod502/tidier) because it is labeled with "waiting on response" and has not seen any activity for 90 days. But don't worry—if you have any information that might advance the discussion, leave a comment and the thread may be reopened :)
